src/main/webapp/app/iris/base-chatbot/iris-base-chatbot.component.ts

```diff
@@ -114,16 +120,16 @@ export class IrisBaseChatbotComponent implements OnInit, OnDestroy, AfterViewIni
     resendAnimationActive: boolean;
     public ButtonType = ButtonType;
 
-    @Input() fullSize: boolean | undefined;
-    @Input() showCloseButton: boolean = false;
-    @Output() fullSizeToggle = new EventEmitter<void>();
-    @Output() closeClicked = new EventEmitter<void>();
+    fullSize = input<boolean>();
+    showCloseButton = input<boolean>(false);
+    fullSizeToggle = output<void>();
+    closeClicked = output<void>();
 
-    // ViewChilds
-    @ViewChild('messagesElement') messagesElement!: ElementRef;
-    @ViewChild('scrollArrow') scrollArrow!: ElementRef;
-    @ViewChild('messageTextarea') messageTextarea: ElementRef<HTMLTextAreaElement>;
-    @ViewChild('acceptButton') acceptButton: ElementRef<HTMLButtonElement>;
+    // ViewChildren
+    messagesElement = viewChild<ElementRef>('messagesElement');
+    scrollArrow = viewChild<ElementRef>('scrollArrow');
+    messageTextarea = viewChild<ElementRef<HTMLTextAreaElement>>('messageTextarea');
+    acceptButton = viewChild<ElementRef<HTMLButtonElement>>('acceptButton');
```

Comment:
optional: here if you sure that inputs will be required or a viewChild will most definitely be there. (This is of course not possible if e.g. an element is hidden at first)
```suggestion
    fullSize = input.required<boolean>();
    showCloseButton = input<boolean>(false);
    fullSizeToggle = output<void>();
    closeClicked = output<void>();

    // ViewChildren
    messagesElement = viewChild.required<ElementRef>('messagesElement');
    scrollArrow = viewChild.required<ElementRef>('scrollArrow');
    messageTextarea = viewChild.required<ElementRef<HTMLTextAreaElement>>('messageTextarea');
    acceptButton = viewChild.required<ElementRef<HTMLButtonElement>>('acceptButton');
```

Reply:
fullsize must be optional, since no button will be displayed if fullsize is undefined. But I added required for the viewchildren